File: src/leetQ58/Solution.java
package leetQ58;

/**
 * Created by zhangzhongwen on 2019/1/10
 */
public class Solution {
    public static void main(String[] args) {
        System.out.println(lengthOfLastWord("Hello World"));
        System.out.println(lengthOfLastWord("a"));
    }

    public static int lengthOfLastWord(String s) {
        String[] words = s.split("[ //,//?//!]");
        if (words.length == 0) return 0;
        return words[words.length - 1].length();
    }
}

File: src/leetQ58/Solution.kt
package leetQ58

/**
 * Created by zhangzhongwen on 2019/1/10
 */

fun main(args: Array<String>) {
    println(lengthOfLastWord("Hello World"))
    println(lengthOfLastWord("a"))
}

fun lengthOfLastWord(s: String): Int {
    val words = s.split("[ //,//?//!]".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
    return if (words.isEmpty()) 0 else words[words.size - 1].length
}